internal/configs/virtualserver.go

```diff
@@ -264,20 +264,30 @@ func generateProxyPassProtocol(upstream conf_v1alpha1.Upstream) string {
 	return "http"
 }
 
+func generateStringOrDefault(s string, defaultS string) string {
```

Comment:
this function needs a unit test

Reply:
Additionally, other similar functions don't have "OrDefault" suffix. Please remove it.